server/src/testFixtures/java/io/crate/operation/aggregation/AggregationTestCase.java

```diff
@@ -175,6 +175,101 @@ public Object executeAggregation(Signature boundSignature,
         );
     }
 
+    public Object executeAggregationWithDocValues(Signature maybeUnboundSignature,
```

Comment:
The reason for these two test methods is that in the topk case the partial result TopKState can not be used for assertion, only the final result.  `AggregationTestCase.executeAggregation` executes and tests against the partial results internally for doc-values and non-doc-values and this is not applicable here.

Reply:
> the partial result TopKState can not be used for assertion

Why not?

Would be great if we could avoid adding additional methods to the `AggregationTestCase`

Reply:
The reason is that the Sketch data structure is not comparable. This makes sense because of the internal state of top-k algorithm which cannot really be compared, because of it's statistics. We first need to run `terminatePartial` to fetch the top-k hits and then comapre.

Reply:
Couldn't we either:

1. Implement a custom equals/hashCode on `TopKState` that's doing an implicit terminate
2. Parameterize a `compare` to `executeAggregation`
3. Parameterize if comparison should be done on partial or final result
4. Generally change it to compare on final result

?

Reply:
Thanks for the feedback. I made the TopK comparable.